src/dune_rules/lib.ml

```diff
@@ -946,10 +947,9 @@ end = struct
         let res =
           let open Resolve.Memo.O in
           let* vlib = resolve_forbid_ignore name in
-          let virtual_ = Lib_info.virtual_ vlib.info in
-          match virtual_ with
-          | false -> Error.not_virtual_lib ~loc ~impl:info ~not_vlib:vlib.info
-          | true -> Resolve.Memo.return vlib
+          match Lib_info.kind vlib.info with
+          | Dune_file _ -> Error.not_virtual_lib ~loc ~impl:info ~not_vlib:vlib.info
+          | Parameter | Virtual -> Resolve.Memo.return vlib
```

Comment:
`Parameter` cannot be accepted here.

Reply:
At this point in the code we are trying to resolve the Lib_info fields into Lib.t values, so when we look at the `implements` we can actually get either a Virtual or a Parameter (since it's the same field in the library stanza, and we have no way of knowing what it'll be before we resolve the lib name with the database). Should we rename the `vlib` variable to avoid the confusion?

Reply:
Renaming it would be useful. Thanks.

Reply:
Done, I've renamed it to `implements` :)